cachito/web/content_manifest.py

```diff
@@ -85,9 +104,50 @@ def process_go_package(self, package, dependency):
         :param Dependency dependency: the go-package package dependency to process
         """
         if dependency.type == "go-package":
-            icm_dependency = {"purl": to_purl(dependency)}
+            if dependency.version and dependency.version.startswith("."):
+                dep_purl = self._get_local_go_package_dep_purl(package, dependency)
+            else:
+                dep_purl = to_purl(dependency)
+
+            icm_dependency = {"purl": dep_purl}
             self._gopkg_data[package]["dependencies"].append(icm_dependency)
 
+    def _get_local_go_package_dep_purl(self, package: "Package", dependency: "Package") -> str:
+        """
+        Get the purl for a local go-package dependency.
+
+        :param Package package: the go-package package to process
+        :param Dependency dependency: the go-package package dependency to process
+        """
+        if not dependency.version or not dependency.version.startswith("."):
+            raise ValueError("{dependency} has an invalid version for a local dependency")
+
+        modules = {module.name: module for module in self.packages if module.type == "gomod"}
+        dep_module_name = gomod.match_parent_module(dependency.name, modules.keys())
+
+        # if the dep_module is in this repo, replace the dependency version with the module version
+        if dep_module_name is not None:
+            dependency.version = modules[dep_module_name].version
+            return to_purl(dependency)
```

Comment:
Ah, but we won't be able to un-special-case it here. The relative path from the dep_module_name to the dep_normpath won't be empty unless the package is at the module root.

In that case, it's probably better to keep the special case for `gomod` as well